### PR TITLE
Updates label and legend styles for Checkbox, Radio, CheckboxGroup, and RadioGroup

### PR DESCRIPTION
**Pull request by mperrotti**
Closes https://github.com/primer/react/issues/5311

#### Before

Screenshot 2024-11-25 at 5 02 11 PM

#### After

Screenshot 2024-11-25 at 5 01 45 PM

#### Changed

- Makes {Checkbox|Radio}Group `<legend>` bold and default font size
- Makes Checkbox and Radio `<label>` default font weight

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/FormControl/FormControl.tsx [hunk 1]
```diff
@@ -172,17 +172,15 @@ const FormControl = React.forwardRef<HTMLDivElement, FormControlProps>(
                 {slots.leadingVisual}
               </Box>
             )}
-            {!slots.label?.props.visuallyHidden || slots.caption ? (
```
Comment (francinelucca) [inline packages/react/src/FormControl/FormControl.tsx:175]: is there a reason we don't need this check anymore?

File: packages/react/src/internal/components/InputLabel.tsx [hunk 2]
```diff
@@ -54,7 +54,6 @@ const InputLabel: React.FC<React.PropsWithChildren<Props>> = ({
         display: 'block',
         color: disabled ? 'fg.muted' : 'fg.default',
         cursor: disabled ? 'not-allowed' : 'pointer',
-        alignSelf: 'flex-start',
```
Comment (francinelucca) [inline packages/react/src/internal/components/InputLabel.tsx:57]: Just making sure this is an intended change? as it doesn't seem directly correlated with Checkbox

**Review by francinelucca — COMMENTED** [review]
LGTM code wise, just a few questions.
Also since this has the potential to touch a lot of areas indirectly, I'd recommend running the integration tests and doing some manual testing in RL to make sure everything still looks good

File: packages/react/src/FormControl/FormControl.tsx (hunk 1 above)
Comment (mperrotti) [reply packages/react/src/FormControl/FormControl.tsx:175]: I couldn't figure out why needed this check, and removing it doesn't break any VRTs or unit tests.

Comment (francinelucca) [reply packages/react/src/FormControl/FormControl.tsx:175]: Did some local testing and couldn't force a difference either. Went down a bit of a rabbit hole and saw you [authored](https://github.com/primer/react/pull/1836/files#diff-868bfd75f5bcb508e151876612d32055aa262949ee9c675eed2187efc311ec08R164) this change so I'll trust you, approving! Let me know if you want help running integration checks or testing in RL

**Review by francinelucca — APPROVED** [review]

**Pull request merged**